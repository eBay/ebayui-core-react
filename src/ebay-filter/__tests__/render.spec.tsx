import React from 'react'
import { composeStories } from '@storybook/react'

import * as stories from './index.stories'
import { render } from '@testing-library/react'
import EbayFilter from '../filter'

const { Default } = composeStories(stories)

describe('<EbayFilter /> rendering', () => {
    it('renders default story correctly', () => {
        const { container } = render(<Default />)
        expect(container).toMatchSnapshot()
    })

    it('renders with pressed attribute', () => {
        const { container } = render(<EbayFilter selected useAriaPressed>text</EbayFilter>)
        expect(container).toMatchSnapshot()
    })

    it('renders without pressed attribute', async () => {
        const { container } = render(<EbayFilter selected useAriaPressed={false}>text</EbayFilter>)
        expect(container).toMatchSnapshot()
    });

    it('renders with disabled attribute', async () => {
        const { container } = render(<EbayFilter disabled>text</EbayFilter>)
        expect(container).toMatchSnapshot()
    });

    it('renders with link', async () => {
        const { container } = render(<EbayFilter href="#link">text</EbayFilter>)
        expect(container).toMatchSnapshot()
    });

    it('renders with link and disabled attribute', async () => {
        const { container } = render(<EbayFilter href="#link" disabled>text</EbayFilter>)
        expect(container).toMatchSnapshot()
    });
})
