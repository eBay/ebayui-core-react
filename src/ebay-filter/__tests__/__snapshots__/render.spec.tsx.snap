// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EbayFilter /> rendering renders default story correctly 1`] = `
<div>
  <button
    class="filter-button filter-button--unselected"
    type="button"
  >
    <span
      class="filter-button__cell"
    >
      <span>
        Filter
      </span>
    </span>
  </button>
</div>
`;

exports[`<EbayFilter /> rendering renders with disabled attribute 1`] = `
<div>
  <button
    class="filter-button filter-button--unselected"
    disabled=""
    type="button"
  >
    <span
      class="filter-button__cell"
    >
      <span>
        text
      </span>
    </span>
  </button>
</div>
`;

exports[`<EbayFilter /> rendering renders with link 1`] = `
<div>
  <a
    class="filter-link filter-link--unselected"
    href="#link"
  >
    <span
      class="filter-link__cell"
    >
      <span>
        text
      </span>
    </span>
  </a>
</div>
`;

exports[`<EbayFilter /> rendering renders with link and disabled attribute 1`] = `
<div>
  <a
    class="filter-link filter-link--unselected"
    disabled=""
    href="#link"
  >
    <span
      class="filter-link__cell"
    >
      <span>
        text
      </span>
    </span>
  </a>
</div>
`;

exports[`<EbayFilter /> rendering renders with pressed attribute 1`] = `
<div>
  <button
    aria-pressed="true"
    class="filter-button filter-button--selected"
    type="button"
  >
    <span
      class="filter-button__cell"
    >
      <span>
        text
      </span>
    </span>
  </button>
</div>
`;

exports[`<EbayFilter /> rendering renders without pressed attribute 1`] = `
<div>
  <button
    class="filter-button filter-button--selected"
    type="button"
  >
    <span
      class="filter-button__cell"
    >
      <span>
        text
      </span>
    </span>
  </button>
</div>
`;
